### PR TITLE
Fix WebP memory leaks

### PR DESCRIPTION
**Pull request by benoit-pierre**
- add a test that will not be run automatically: `Tests/check_webp_leaks.py`
- fix 2 memory leaks
- improve error checking when using the Mux API

This fixes issue #586.

### CODE REVIEW

File: Tests/check_webp_leaks.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+from helper import unittest, PillowTestCase
+import sys
+from PIL import Image
+from io import BytesIO
+
+# Limits for testing the leak
+mem_limit = 16 # max increase in Mo
```
Comment (hugovk) [inline Tests/check_webp_leaks.py]: "Mo" is "MB" in English :)

**Comment by hugovk** [comment]
How long, roughly, does the test take to run? Too long to be included on the CI?

**Comment by benoit-pierre** [comment]
> "Mo" is "MB" in English :)

Oups... Démasqué...

> How long, roughly, does the test take to run? Too long to be included on the CI?

Not long: ~2.5s on my machine (i7-3770 @ 3.40GHz). The problem is whether the method is reliable on different configurations.

And I just noticed the test class name is wrong...

**Comment by benoit-pierre** [comment]
Pushed an update with an amended test:
- "Mo" -> "MB"
- fix class name
- fix Python 3 support

**Comment by hugovk** [comment]
I tested it on the CI and it passed for all configurations there:
https://travis-ci.org/hugovk/Pillow/builds/52132563

@wiredfool What do you think? Should this test be part of the main test suite?

**Comment by wiredfool** [comment]
No, lets leave potential DOS out of the test suite. I still run this suite on a Pi B and a 1.2 Ghz g4.

**Pull request merged**